defos/src/defos_html5.cpp

```diff
@@ -144,6 +144,36 @@ void defos_set_window_title(const char* title_lua) {
     EM_ASM_({document.title = UTF8ToString($0)}, title_lua);
 }
 
+void defos_set_window_icon(const char *icon_path)
+{
+    EM_ASM_({
+        document.head || (document.head = document.getElementsByTagName('head')[0]);
+        function changeFavicon(src) {
+            var link = document.createElement('link');
+            var oldLink = document.getElementById('dynamic-favicon');
+            link.id = 'dynamic-favicon';
+            link.rel = 'shortcut icon';
+            link.href = src;
+            if (oldLink) {
+                document.head.removeChild(oldLink);
+            }
+            document.head.appendChild(link);
+        }
+        changeFavicon(UTF8ToString($0));
+    }, icon_path);
+}
+
+char const* defos_get_bundle_root() {
+    char*bundlePath = (char*)EM_ASM_INT({
+        var jsString = location.href.substring(0, location.href.lastIndexOf("/"));
+        var lengthBytes = lengthBytesUTF8(jsString)+1; // 'jsString.length' would return the length of the string as UTF-16 units, but Emscripten C strings operate as UTF-8.
+        var stringOnWasmHeap = _malloc(lengthBytes);
+        stringToUTF8(jsString, stringOnWasmHeap, lengthBytes+1);
+        return stringOnWasmHeap;
+    },0);
+    return bundlePath;
```

Comment:
This leaks the `bundlePath` string after it gets passed to Lua in `get_bundle_root()`. Maybe always make sure we return a string we own and `free()` it in `get_bundle_root()`?

Reply:
I wanted to do this (I asked you aboutt free()), it was my fault with conts, thx!

Reply:
done